source/DataNode.cpp

```diff
@@ -286,9 +317,45 @@ int DataNode::PrintTrace(const string &message) const
 
 
 
+// Get the value, and if it was a variable, the variable name
+Condition<double> DataNode::AsCondition(int index) const
+{
+	// Check for empty strings and out-of-bounds indices.
+	if(static_cast<size_t>(index) >= tokens.size() || tokens[index].empty())
+		PrintTrace("Error: Requested token index (" + to_string(index) + ") is out of bounds:");
+	else if(IsNumber(tokens[index]))
+		return Condition<double>(Value(index));
+	else if(IsCondition(tokens[index]))
+		return Condition<double>(store, tokens[index].substr(1));
+	PrintTrace("Cannot convert value \"" + tokens[index] + "\" to a number or condition name. "
+		"Condition names must begin with a dollar sign (\"&\"). "
```

Comment:
you forgot to update this
the & should be a global constant? make it a static member of Condition maybe?

Reply:
I think it belongs in a DataNode static constant since Condition is just a reference; it should not know how it is read in (or even if it is read in). The DataNode is the only class that ever reads a Condition, and ConditionSet already knows about DataNode.

Reply:
The name of the condition char is now in DataNode::CONDITION_CHAR_NAME which calls it an ampersand.